app/server.js

```diff
@@ -47,26 +47,13 @@ async function startServer() {
     const { app, ipfs } = await createHttpServer()
 
     const server = await new Promise((resolve, reject) => {
-      let resolved = false
       const server = app.listen(PORT, (err) => {
-        if (err) {
-          if (!resolved) {
-            resolved = true
-            reject(err)
-          }
-        }
+        if (err) return reject(err)
```

Comment:
can this lead to reject being called twice?

Reply:
no, it will stop and return reject, basically code below will be ignored if (err) translates to true. As far as I'm aware of promise should only call once resolve or either reject

Reply:
the return statement